setuptools/command/egg_info.py

```diff
@@ -384,10 +384,10 @@ def process_template_line(self, line):
 
         try:
             process_action = action_map[action]
-        except KeyError:
+        except KeyError as e:
             raise DistutilsInternalError(
                 "this cannot happen: invalid action '{action!s}'".format(action=action),
```

Comment:
I think something like this would make it easier to identify and correct mistakes:

```suggestion
                f"Invalid MANIFEST.in: unknow action {action!r} in {line!r}"
```

But I am not sure if this change requires further changes in the tests...

Reply:
Let's give it a try... I should be able to modify the tests accordingly.